# Best Model Selection - Summary Table of Trained Models
This Table compares the R² values of the 6 trained Models, and finds out which Model is most suitable out of all for implementation. Here only the best performing variation of the Model will be compared with each other Model.

In [2]:
# Import necessary libraries
import pandas as pd

# Results of each model
results = [
    ["Decision Tree", 0.517094],
    ["Random Forest", 0.5546],
    ["MLP", 0.5533],
    ["XGBoost", 0.566784],
    ["SVR", 0.5147435174768522],
    ["Linear Regression", 0.51897]
]

# Create DataFrame
results_df = pd.DataFrame(results, columns=["Model", "R²"])

# Display the table
display(results_df)

# Print the best model based on R²
best_model_row = results_df.loc[results_df['R²'].idxmax()]
print(f"Best model: {best_model_row['Model']} (R²={best_model_row['R²']:.4f})")

Model        R²
0      Decision Tree  0.517094
1      Random Forest  0.554600
2                MLP  0.553300
3            XGBoost  0.566784
4                SVR  0.514744
5  Linear Regression  0.518970

Best model: XGBoost (R²=0.5668)


# Performance we Observed
This is what we can deduce from the results of the comparison
- The XGBoost model has the highest R² value (0.5668), indicating it explains the most variance in the test data among all models.
- Random Forest and MLP also perform reasonably well, with R² values above 0.55.
- Decision Tree, SVR, and Linear Regression have lower R² values, indicating less predictive power.

# Challenges we faced
These are the Challenges we faced while training our Models
- Data Quality Issues
- Feature Selection
- Imbalanced Data
- Hyperparameter Tuning
- Overfitting
- Model Interpretability
- Computational Resources
- Evaluation Consistency
- Deployment Considerations

# Behaviours we Expected
These are the Behaviours we Expected while training our Models
- No Overfitting & Underfitting
- Low Error Metrics
- Stable Performance
- Reasonable Feature Importance
- No Major Prediction Bias
- Efficient Computation
- Interpretable Results
- Robustness to Outliers
- Good Generalization